lib/services/plugins-service.ts

```diff
@@ -0,0 +1,190 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import shelljs = require("shelljs");
+import semver = require("semver");
+import Future = require("fibers/future");
+import constants = require("./../constants");
+let xmlmerge = require("xmlmerge-js");
+
+export class PluginsService implements IPluginsService {
+	private static INSTALL_COMMAND_NAME = "install";
+	private static UNINSTALL_COMMAND_NAME = "uninstall";
+	
+	constructor(private $npm: INodePackageManager,
+		private $fs: IFileSystem,
+		private $projectData: IProjectData,
+		private $platformsData: IPlatformsData,
+		private $projectDataService: IProjectDataService,
+		private $childProcess: IChildProcess,
+		private $options: IOptions,
+		private $logger: ILogger,
+		private $errors: IErrors) { }
+		
+	public add(plugin: string): IFuture<void> {
+		return (() => {
+			let pluginName = this.executeNpmCommand(PluginsService.INSTALL_COMMAND_NAME, plugin).wait();
+			let nodeModuleData = this.getNodeModuleData(pluginName);
+			if(!nodeModuleData.isPlugin) {
+				// We should remove already downloaded plugin and show an error message
+				this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+				this.$errors.failWithoutHelp(`The specified plugin ${plugin} is not a valid NativeScript plugin. Ensure that the plugin contains nativescript key in package.json file and try again.`);
+			}
+			
+			this.prepare(this.convertToPluginData(nodeModuleData)).wait();
+			this.$logger.out(`Succsessfully installed plugin with name ${nodeModuleData.name}.`);	
+		}).future<void>()();
+	}
+	
+	public remove(pluginName: string): IFuture<void> {
+		return (() => {
+			this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+			let showMessage = true;
+			let action = (modulesDestinationPath: string, platform: string, platformData: IPlatformData) => {
+				shelljs.rm("-rf", path.join(modulesDestinationPath, pluginName));
+				this.$logger.out(`Successfully removed plugin ${pluginName} for ${platform} platform`);
+				showMessage = false;
+			};
+			this.executeForAllInstalledPlatforms(action);
+			
+			if(showMessage) {
+				this.$logger.out(`Succsessfully removed plugin ${pluginName}`);	
+			}
+		}).future<void>()();
+	}
+	
+	public prepare(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let action = (pluginDestinationPath: string, platform: string, platformData: IPlatformData) => {
+				let skipExecution = false;
+				// Process .js files 				
+				let installedFrameworkVersion = this.getInstalledFrameworkVersion(platform).wait();
+				let pluginVersion = (<any>pluginData.platformsData)[platform];
+				if(semver.gt(pluginVersion, installedFrameworkVersion)) {
+					this.$logger.warn(`Plugin ${pluginData.name} with specified version ${pluginVersion} for ${platform} platform is not compatible for currently installed framework with version ${installedFrameworkVersion}.`);
```

Comment:
${pluginData.name} ${pluginVersion} for ${platform} is not compatible with the currently installed framework version ${installedFrameworkVersion}.